src/HW_models/NRF_RTC.c

```diff
@@ -125,15 +124,52 @@ static void time_of_1_counter_wrap(int rtc, bs_time_t *us, uint16_t *frac) {
   *us = counter_to_time((uint64_t)RTC_COUNTER_MASK + 1, rtc, frac);
 }
 
+/**
+ * Convert a counter value to absolute time.
+ * @note To obtain a time delta, use @ref counter_to_time instead
+ */
+static bs_time_t counter_to_abs_time(int rtc, uint32_t rtc_counter) {
+  uint16_t frac;
+  bs_time_t bs_time = RTC_counter_startT[rtc]
+                      + counter_to_time(rtc_counter, rtc, &frac);
+
+  bs_time_t now = tm_get_hw_time();
+
+  if (bs_time <= now) {
+    bs_time_t t_us;
+    uint16_t t_frac;
+
+    time_of_1_counter_wrap(rtc, &t_us, &t_frac);
+
+    do {
+      bs_time += t_us;
+      frac += t_frac;
+      if (frac >= 0x200){
+        frac -= 0x200;
+        bs_time +=1;
+      }
+    } while (bs_time <= now);
+  }
+
+  return bs_time;
+}
+
 static void update_master_timer() {
   Timer_RTC = TIME_NEVER;
   for (int rtc = 0; rtc < N_RTC ; rtc++) {
     if (RTC_Running[rtc] == false) {
       continue;
     }
-    for (int cc = 0 ; cc < N_CC ; cc++) {
-      if (cc_timers[rtc][cc] < Timer_RTC) {
-        Timer_RTC = cc_timers[rtc][cc];
+
+    if (NRF_RTC_regs[rtc].EVTEN & RTC_INTENSET_TICK_Msk) {
+      // The next event happens at the next counter tick
+      nrf_rtc_update_COUNTER(rtc);
+      Timer_RTC = counter_to_abs_time(rtc, (NRF_RTC_regs[rtc].COUNTER + 1) & RTC_COUNTER_MASK);
+    } else {
+      for (int cc = 0 ; cc < N_CC ; cc++) {
+        if (cc_timers[rtc][cc] < Timer_RTC) {
+          Timer_RTC = cc_timers[rtc][cc];
+        }
```

Comment:
Should this really be in an "else" branch? What if this would calculate an earlier time than was calculated in the "if" branch?

Reply:
Good question. The earliest possible time a new CC can trigger is COUNTER + 1, which is also the time where EVENTS_TICK will trigger. Therefore we do not really need to check the CC values.

Reply:
As pointed out in https://github.com/BabbleSim/ext_NRF52_hw_models/pull/16#discussion_r526729183, this was actually wrong. In case you are running multiple RTCs with different prescale values, a CC from another RTC can trigger earlier that the tick event. Now the implementation is updated to handle this case.